BP-320: Display predictions in chart

julianweise opened the pull request:
Implemented bar chart (used CharJS) to display all causes and their probability of a specific errorType.
All errorTypes are displayed within a product as list. Clicking on an list element opens a new view (modal) to display the bar chart.
Attention: You have to select a configuration to view the list of errorTypes!

MaximilianV committed:
add second header for choosing product configuration

MaximilianV committed:
save selection of SW and CP in state

MaximilianV committed:
fix missmatching state values

MaximilianV committed:
add callback for configuration to productView

MaximilianV committed:
move product header up and back to header

MaximilianV committed:
move configuration header css up to header

MaximilianV committed:
move configuration data mock up to productview

friedow committed:
Rebased and merged config

MaximilianV committed:
adapt configuration selection to api

MaximilianV committed:
adjust RemoteHandler to fit configurations

MaximilianV committed:
show status derivation of configurations

julianweise committed:
Added SeverMock Data and testData for configurations

julianweise committed:
Fixed problem in case of not defined configurations

julianweise committed:
Implemented List of ErrorTypes in ProductView

julianweise committed:
Ignored config

julianweise committed:
Solved rebase problem of doublicated method

julianweise committed:
Added model for each prediction

julianweise committed:
Implemented chart

julianweise committed:
Added real data to prediction view chart

julianweise committed:
Improved errorType / display code mock data

julianweise committed:
Enlarged modal view for prediction view

julianweise committed:
Improved styling of prediction view modal (enlarged view size)

julianweise committed:
Removed unused method

julianweise committed:
Removed unnecessary import in test

julianweise committed:
Added test for predictionList

julianweise committed:
Updated tests and related snapshots for predictionview and -tests

julianweise committed:
Removed PredictionView from tests

This is possible due to the fact that ReactTestUtils/Jest does not support canvas in its test environments.

hpibptlab commented on src/__tests__/ProductView/PredictionList.js:
```diff
@@ -0,0 +1,13 @@
+    const component = renderer.create(
+        <PredictionList
+            configuration={testConfigurations.CONFIGURATIONS[0]} />);
+    let tree = component.toJSON();
```
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Make "tree" "const". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://bpt-lab.org/sonarqube/coding_rules#rule_key=javascript%3AS3353)

hpibptlab commented:
SonarQube analysis reported 1 issue
* ![MINOR][MINOR] 1 minor

Watch the comments in this conversation to review them.

[MINOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR'

julianweise committed:
Fixed code issue

Changed let type tp const type

tommartensen commented on src/__tests__/testData/frontend_configurations.js:
```diff
@@ -13,96 +13,102 @@ const configurations =   [{
+            errorDescription: "Heating produces strange sounds",
+            errorCauses: [{
+                id: 0,
+                causeDescription: "Kabelbaum missing",
```
Please translate this into english!

tommartensen commented on src/__tests__/testData/frontend_configurations.js:
```diff
@@ -13,96 +13,102 @@ const configurations =   [{
+            errorDescription: "Heating produces strange sounds",
+            errorCauses: [{
+                id: 0,
+                causeDescription: "Kabelbaum missing",
```
Please translate this into english!

tommartensen commented on src/__tests__/testData/frontend_configurations.js:
```diff
@@ -13,96 +13,102 @@ const configurations =   [{
+            errorDescription: "Heating produces strange sounds",
+            errorCauses: [{
+                id: 0,
+                causeDescription: "Kabelbaum missing",
```
Please translate this into english!

friedow commented on src/ProductView/Header/Header.js:
```diff
@@ -4,31 +4,26 @@ import ConfigurationHeader from './ConfigurationHeader/ConfigurationHeader.js';
                             </div>
-                            <div className="col-12 d-flex justify-content-center">
+                            <div className="col-10 d-flex justify-content-center">
                                 <ConfigurationHeader configurations={this.props.configurations} onChangeProductConfiguration={this.props.onChangeProductConfiguration}/>
```
Line too long.

julianweise replied:
fixed

friedow commented on src/ProductView/PredictionList/PredictionList.js:
```diff
@@ -0,0 +1,30 @@
+
+class PredictionList extends Component {
+
+    constructor(props) {
```
Remove unnecessary constructor.

julianweise replied:
solved

friedow commented on src/ProductView/PredictionList/PredictionList.js:
```diff
@@ -0,0 +1,30 @@
+    }
+
+    render() {
+        const errorList = []; 
```
Please extract the preparation of errorList into a own method.

julianweise replied:
done

friedow commented on src/ProductView/PredictionList/PredictionList.scss:
```diff
@@ -0,0 +1,11 @@
+@import "PredictionView/PredictionView";
+
+.prediction-list-container {
+  margin-top: 2vw;
```
Is using vw safe here?

julianweise replied:
Changed it to px.

friedow commented on src/ProductView/PredictionList/PredictionList.scss:
```diff
@@ -0,0 +1,11 @@
+
+.prediction-list-container h2 {
+  margin-bottom: 10px;
+  width: 100%;
```
please use a w-100 css class onto the html element instead of using width: 100% here.
Reference: https://v4-alpha.getbootstrap.com/utilities/sizing/

julianweise replied:
fixed

friedow commented on src/ProductView/PredictionList/PredictionView/PredictionView.js:
```diff
@@ -0,0 +1,93 @@
+    }
+
+    render() {
+        const modalId = "#prediction-view-"+this.props.errorType.errorTypeId+"-modal";
```
You should extract the id selector into a const variable.

julianweise replied:
fixed

friedow commented on src/ProductView/ProductView.js:
```diff
@@ -214,6 +215,10 @@ class ProductView extends Component {
                             </div>
                             <div className="col-12">
+                                <PredictionList
+                                    configuration={this.state.configuration} />
```
Don't break that.

julianweise replied:
outdated, now has to props.

friedow commented on src/__tests__/ProductView/PredictionList.js:
```diff
@@ -0,0 +1,13 @@
+test('Render DetailArea', () => {
+    const component = renderer.create(
+        <PredictionList
+            configuration={testConfigurations.CONFIGURATIONS[0]} />);
```
Don't break that.

julianweise replied:
see comment above.

julianweise committed:
Translated testData

julianweise committed:
Shorten too long line

julianweise committed:
Outsourced generation of errorCode-list to separate method

julianweise committed:
Used px instead of relative vw for margin

julianweise committed:
Fixed bug of not showing error-code list

julianweise committed:
Fixed display bug of badget in error-code list

julianweise committed:
Conditional rendering depending on showAllConfigurations

In case of displaying all configurations error-code list should not be displayed. Furthermore, added tests and updated snapshots.

julianweise committed:
Conditional rendering updates

julianweise committed:
Translated description in server mock data

julianweise committed:
Used bootstrap class instead of css attribute.

julianweise committed:
Extracted state selector to separate const variable

julianweise committed:
Fixed typo

julianweise committed:
Changed rendering in case of showing all configurations

Also adapted tests and updated snapshots.
